draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+While {{?RFC8288}} web linking solves this problem in a more general way, a simplified, concise, and self-contained linking and identification system is better-suited to many specifications.  This allows those specifications to avoid complexity, and to avoid constraining their syntax to match any one of the many competing web linking-enabled JSON formats.
+
+To support a variety of use cases, JRI supports trade-offs between generic interoperability and tight integration with the context in which it is used.  JRI defines an interoperable subset of its features for standalone use, while encouraging specifications that need reference and/or identification features to incorporate as much or as little of JRI as suits their needs.  JRI also defines mechanisms for resources to indicate their JRI usage, allowing generic or standalone implementations to understand how to safely process a document without knowledge of its underlying data format.
+
+## Notational Conventions
+
+{::boilerplate bcp14+}
+
+The terms "primary resource" and "secondary resource" in this specification are to be interpreted as in {{!RFC3986, Section 3.5}}.
+
+Due to the distinction that this specification makes between documents and resources, the term "same-resource reference" is to be interpreted in the same way as "same-document reference" as defined in {{!RFC3986, Section 4.4}}.  The term "same-document reference" is not used in this specification.
+
+The terms "IRI", "IRI-reference", "relative reference", "absolute-IRI", and "base IRI" in this specification are to be interpreted as in {{!RFC3987}}.
+
+## Definitions
+
+In addition to the terms from other specifications listed in the previous section, this document uses the following terminology:
+
+* _data format:_ any specification for structuring data within a document, including but not limited to formal media types
+* _JSON-based format:_ any data format using {{?RFC8259}} JSON as its syntax, including but not limited to formats defined with an `application/json` or `+json` structured suffix media type
+* _JSON Pointer-compatible document:_ any document with a structure that can be unambiguously addressed by JSON Pointer as described in {{jsonpointer}}
+* _root object:_ the object, if any, containing the entirety of a JSON-compatible document[^2]
+* _keyword:_ an object property name with specification-defined behavior, as opposed to object properties which are simply data
+* _processing a data format:_ performing any actions required by the format, typically based on the format's structure and keywords, beyond simply parsing the format into an in-memory representation
+* _evaluation path:_ the path, including reference keywords traversed, taken to reach a particular point of processing from the document root at which processing started; the exact nature of this path depends on the reference semantics chosen as defined in {{ref-semantics}}
```

Comment:
There are non-schema JSON Reference tools that use an evaluation path concept, although they don't use that exact name for it.  This is why the JRI spec talks about whether `$ref` appears in the evaluation path (the approach of not showing it is what at least one very popular tool does).

"Keyword" is to distinguish object property names that _do_ something vs those that don't, and yes other specifications besits JSON Schema have keywords in this sense, or are "processed."  Depending on what you're doing with them, OpenAPI and AsyncAPI documents are processed.

Document root and root object are separate because it's not yet clear if documents with a non-object root are something that need to be supported.  If not, then document root and root object will end up being the same thing.